src/query/remote/client.go

```diff
@@ -324,5 +424,6 @@ func (c *grpcClient) CompleteTags(
 }
 
 func (c *grpcClient) Close() error {
+	close(c.closeCh)
```

Comment:
any race issue with the health checker potentially using the closed connection? probably not a big deal, maybe an error gets generated

Reply:
True, I can probably do a better job here with sync.rwmutex protecting closed state and read that under RLock before attempting health check in the connection.